aws/networking/main.tf

```diff
@@ -0,0 +1,79 @@
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+resource "aws_vpc" "main" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+  tags = local.tags
+}
+
+resource "aws_internet_gateway" "main" {
+  vpc_id = aws_vpc.main.id
+  tags = local.tags
+}
+
+resource "aws_egress_only_internet_gateway" "main" {
+  vpc_id = aws_vpc.main.id
+  tags = local.tags
+}
+
+resource "aws_route_table" "main" {
+  vpc_id = aws_vpc.main.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.main.id
+  }
+
+  route {
+    ipv6_cidr_block        = "::/0"
+    egress_only_gateway_id = aws_egress_only_internet_gateway.main.id
+  }
+  tags = local.tags
+}
+
+resource "aws_main_route_table_association" "default" {
+  vpc_id         = aws_vpc.main.id
+  route_table_id = aws_route_table.main.id
+}
+
+resource "aws_subnet" "primary" {
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = "10.0.1.0/24"
```

Comment:
Input with default value here as well.

Reply:
I suggest shipping these as-is, as presets and use the `cidrsubnet`[^1] function for default values. It will be more difficult to make this both configurable, along with the VPC CIDR and provide out of the box working values.

Let's stick with non-configurable subnets for now and we will think about extensibility in the future, if needed.

[^1]: https://developer.hashicorp.com/terraform/language/functions/cidrsubnet